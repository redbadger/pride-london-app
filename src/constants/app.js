// @flow
export const appNameIos = "pride-in-london";
export const appIdIos = "1250496471";
