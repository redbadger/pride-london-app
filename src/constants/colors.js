// ZEPLIN EXPORT

export const greyishBrownColor = "#585858";
export const lightNavyBlueColor = "#2d2f7f";
export const eucalyptusGreenColor = "#2cda9d";
export const darkBlueGreyColor = "#191a47";
export const darkBlueGreyTwoColor = "#20215a";
export const lightishGreyColor = "#d8d8d8";
export const lightGreyColor = "#f3f3f3";
export const brightLightBlueColor = "#2acfff";
export const darkSkyBlueColor = "#2c77e5";
export const turquoiseBlueColor = "#00a3bd";
export const lightTealColor = "#7de2d1";
export const vomitYellowColor = "#c3d30e";
export const yellowColor = "#ffd95e";
export const warmPinkColor = "#fc4984";
export const bubblegumPinkColor = "#ff88f1";
export const brightPurpleColor = "#bc01ff";
export const cornflowerBlueColor = "#5770d1";
export const whiteColor = "#ffffff";
export const velvetColor = "#660c63";
export const blackColor = "#111111";
export const coralColor = "#ff5b44";
export const shadowColor = "rgba(0, 0, 0, 0.1)";

// CUSTOM

export const bgColor = "#ffffff";
export const cardBgColor = "#ffffff";
export const imageBgColor = "#C1C1C1";
export const headerBgColor = "#A5A5A5";
export const transparent = "transparent";

export const textColor = "#272727";

export const dialogBackdropColor = "rgba(0, 0, 0, 0.7)";
export const blackZeroColor = "rgba(0, 0, 0, 0)";
export const blackFifteenColor = "rgba(0, 0, 0, 0.1)";
export const blackThirtyColor = "rgba(0, 0, 0, 0.3)";
export const whiteZeroColor = "rgba(255, 255, 255, 0)";
export const dialogBgColor = whiteColor;
export const dialogTitleColor = "rgb(45, 47, 127)";
export const dialogApplyButtonBgColor = "rgb(44, 218, 157)";
export const dialogApplyButtonTextColor = "rgb(45, 47, 127)";
export const dialogHeaderDividerColor = "#F2F2F2";
export const disabledButtonBgColor = "#cecece";

// TAB BAR

export const tabBarBgColor = "rgb(243, 243, 243)";
export const tabBarLabelColor = "rgb(88, 88, 88)";
export const tabBarActiveLineColor = "rgb(44, 218, 157)";
export const tabBarActiveLabelColor = "rgb(45, 47, 127)";
export const tabBarShadowColor = shadowColor;
export const tabBarBorderColor = "rgba(0, 0, 0, 0.25)";

// FILTER HEADER

export const filterShowMeTextColor = whiteColor;
export const categoriesFilterButtonBgColor = "rgb(25, 26, 71)";
export const filterBgColor = "rgb(45, 47, 127)";
export const interestButtonBgColor = "rgb(32, 33, 90)";
export const interestButtonTextColor = whiteColor;
export const filterButtonBorderColor = "rgb(44, 218, 157)";
export const filterButtonTextColor = "rgb(44, 218, 157)";
export const filterButtonsBgColor = darkBlueGreyTwoColor;
// EVENT LIST

export const eventListHeaderBgColor = lightishGreyColor;
export const eventListHeaderColor = "rgb(51, 51, 51)";
export const eventCardShadow = "rgba(0, 0, 0, 0.2)";
export const sectionHeaderBgColor = lightGreyColor;
export const sectionHeaderShadow = "rgba(0, 0, 0, 0.14)";
export const sectionHeaderTextColor = "rgb(45, 47, 127)";
// DATE FILTER MODAL

export const dateRangePickerTextColor = "rgb(45, 47, 127)";
export const dateRangePickerDotColor = "rgb(44, 218, 157)";
export const dateRangePickerSelectedColor = "rgb(44, 218, 157)";
export const dateRangePickerCTABgColor = "rgb(44, 218, 157)";
export const dateRangePickerDisabledTextColor = "rgb(203, 203, 203)";

// EVENT DETAILS

export const eventDetailsBgColor = whiteColor;
export const eventDetailsHeaderBgColor = lightGreyColor;
export const eventCategoryLabelBgColor = lightishGreyColor;
export const eventDetailsBuyButtonColor = lightishGreyColor;
export const eventDetailsBuyButtonTextColor = whiteColor;

// HOMEPAGE

export const titleTextColor = "rgb(45, 47, 127)";
export const eventTileTextColor = "rgb(45, 47, 127)";

// SUPPORT US

export const supportUsButtonShadow = "rgba(0, 0, 0, 0.2)";

// SPONSOR
export const sponsorLogoBackgroundColor = "rgba(243, 243, 243, 0.5)";
