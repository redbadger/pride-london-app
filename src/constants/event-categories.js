// Ideally the keys here should be consts, but currently these are setup in
// contentful only as the label, hence we need to use this for the key too.
// If we move to realise categories as its own content type then this will
// go away and we will fetch this data.

const categories = {
  "Cabaret and Variety": {
    label: "Cabaret and Variety"
  },
  Community: {
    label: "Community"
  },
  "Talks and Debates": {
    label: "Talks and Debates"
  },
  "Film and Screenings": {
    label: "Film and Screenings"
  },
  "Plays and Theatre": {
    label: "Plays and Theatre"
  },
  "Social and Networking": {
    label: "Social and Networking"
  },
  Nightlife: {
    label: "Nightlife"
  },
  "Exhibition and Tours": {
    label: "Exhibition and Tours"
  },
  "Sports and Activities": {
    label: "Sports and Activities"
  },
  Health: {
    label: "Health"
  },
  Music: {
    label: "Music"
  }
};

export default categories;
