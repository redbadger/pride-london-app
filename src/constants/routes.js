export const EVENT_DETAILS = "EVENT_DETAILS";
export const EVENT_LIST = "EVENT_LIST";
export const FEATURED_EVENT_LIST = "FEATURED_EVENT_LIST";
export const HOME = "HOME";
export const EVENT_CATEGORIES_FILTER = "EVENT_CATEGORIES_FILTER";
export const PARADE = "PARADE";
export const SAVED = "SAVED";
export const SUPPORT_US = "SUPPORT_US";
export const FILTER_MODAL = "FILTER_MODAL";
export const DONATE = "DONATE";
export const SPONSOR = "SPONSOR";
