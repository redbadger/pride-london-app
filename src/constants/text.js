// @flow
const showEvents = (numberOfEvents: number) => {
  if (numberOfEvents <= 0) {
    return "No events";
  }

  if (numberOfEvents === 1) {
    return "Show 1 event";
  }

  return `Show ${numberOfEvents} events`;
};

const showEventsLabel = (numberOfEvents: number) => {
  if (numberOfEvents <= 0) {
    return "Show events (none selected)";
  }

  if (numberOfEvents === 1) {
    return "Show 1 selected event";
  }

  return `Show ${numberOfEvents} selected events`;
};

export default {
  tabHome: "Home",
  tabEvents: "Events",
  tabParade: "Parade",
  tabSaved: "Saved",
  tabSupportUs: "Support us",
  featuredEventListTitle: "Featured events",
  filterByInterest: "All events",
  addFilters: "Add filters",
  filters: "Filters",
  filterPickerApply: showEvents,
  filterPickerApplyLabel: showEventsLabel,
  filterDayPickerTitle: "Select dates",
  filterTimePickerTitle: "Select time",
  filterTitle: "Show me",
  eventDetailsPrice: "From: Â£",
  isFreePrice: "Free",
  eventFromPrice: "From",
  eventDetailsAccessibility: "Accessibility",
  eventDetailsAccessibilityDetails: "Accessibility Details",
  eventDetailsContact: "Contact",
  eventDetailsBuyButton: "Get tickets",
  eventDetailsGenderNeutralToilets: "Gender neutral toilets",
  selectDates: "Select dates",
  eventDetailsAbout: "About",
  eventDetailsSchedule: "Schedule",
  eventDetailsReadMore: "Read more",
  eventDetailsReadLess: "Read less",
  eventDetailsContactEmailSubject: "Pride In London event enquiry",
  eventDetailsContactEmailBody: "Dear event organiser,",
  collapsibleShowMore: "Show more",
  collapsibleShowLess: "Show less",
  supportUsTitle: "Support us",
  supportUsAsIndividual: "As an individual",
  supportUsAsBusiness: "As a business",
  supportUsDonate: "Donate",
  supportUsDonateDescription: "Every donation keeps us marching",
  supportUsShop: "Buy merchandise",
  supportUsShopDescription: "Visit our online shop on The Gay UK",
  supportUsSponsor: "Partner with us",
  supportUsSponsorDescription: "Support the community",
  supportUsVolunteer: "Volunteer",
  supportUsVolunteerDescription: "Passionate about equality? Join us.",
  sponsorTitle: "Partner with us",
  sponsorLevelHeading: "partners",
  sponsorHeadingOurPartners: "Our main 2018 partners",
  sponsorTextOurPartners:
    "A huge thank you to our main partners for their continued support.",
  sponsorHeadingPartnerWithUs: "Partner with us",
  sponsorTextPartnerWithUs:
    "Build your brand, engage your internal network and be part of a global movement to support diversity. To find out more, please contact sponsor@prideinlondon.org.",
  sponsorTextPartnerWithUsList:
    "* Pride in London is one of the top three biggest events in London, attracting an estimated one million visitors across the footprint. \n* Be part of our award winning marketing and advertising campaign. \n* 80% of those surveyed in our 2016 research agreed that they are more likely to think better of a brand if it is supportive of LGBT+ issues.",
  sponsorContactEmailSubject: "Sponsorship enquiry",
  sponsorContactButtonText: "Email to partner with us",
  anyDay: "Any day",
  anyTime: "Any time",
  time: {
    morning: "Morning",
    afternoon: "Afternoon",
    evening: "Evening"
  },
  tags: {
    audience: "Age group",
    price: "Price",
    free: "Show only free events",
    timeOfDay: "Time of day",
    morning: "Morning",
    afternoon: "Afternoon",
    evening: "Evening",
    venueDetails: "Venue options",
    accessibilityOptions: "Accessibility",
    "Gender neutral toilets": "Gender-neutral toilets",
    Indoors: "Indoor",
    Outdoors: "Outdoor",
    "Accessible Toilets": "Accessible toilets",
    "Wheelchair accessibile": "Wheelchair access",
    area: "Area of London"
  },
  cancel: "Cancel",
  zeroSelected: "0 selected",
  showEvents,
  clearAll: "Clear all",
  filterEvents: "Filter Events"
};
