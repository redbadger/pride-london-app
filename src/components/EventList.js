// @flow
import React, { Component } from "react";
import { StyleSheet, SectionList, View } from "react-native";
import type { SectionBase } from "react-native/Libraries/Lists/SectionList";
import formatDate from "date-fns/format";
import { concat, equals } from "ramda";

import ContentPadding from "./ContentPadding";
import EventCard from "./EventCard";
import Touchable from "./Touchable";
import SectionHeader from "./SectionHeader";
import { selectEventIsFree } from "../selectors/event";
import type { SavedEvents, Event, EventDays } from "../data/event";
import type { LocalizedFieldRef } from "../data/localized-field-ref";
import { bgColor, eventCardShadow } from "../constants/colors";

type Props = {
  locale: string,
  events: EventDays,
  savedEvents: SavedEvents,
  addSavedEvent: string => void,
  removeSavedEvent: string => void,
  refreshing?: boolean,
  onRefresh?: () => void,
  onPress: (eventName: string) => void,
  getAssetUrl: LocalizedFieldRef => string
};

const separator = style => () => <View style={style} />;

type ItemProps = { item: Event };

type RenderItemArgs = {
  isSavedEvent: string => boolean,
  addSavedEvent: string => void,
  removeSavedEvent: string => void,
  locale: string,
  onPress: (eventName: string) => void,
  getAssetUrl: LocalizedFieldRef => string
};

export const renderItem = ({
  isSavedEvent,
  addSavedEvent,
  removeSavedEvent,
  locale,
  onPress,
  getAssetUrl
}: RenderItemArgs) => ({ item }: ItemProps) => (
  <ContentPadding>
    <Touchable
      style={styles.eventListItem}
      onPress={() => onPress(item.sys.id)}
    >
      <EventCard
        name={item.fields.name[locale]}
        locationName={item.fields.locationName[locale]}
        eventPriceLow={item.fields.eventPriceLow[locale]}
        eventPriceHigh={item.fields.eventPriceHigh[locale]}
        startTime={item.fields.startTime[locale]}
        endTime={item.fields.endTime[locale]}
        imageUrl={getAssetUrl(item.fields.eventsListPicture)}
        isFree={selectEventIsFree(item)}
        isSaved={isSavedEvent(item.sys.id)}
        toggleSaved={active => {
          if (active) {
            addSavedEvent(item.sys.id);
          } else {
            removeSavedEvent(item.sys.id);
          }
        }}
      />
    </Touchable>
  </ContentPadding>
);

type Section = SectionBase<Event> & { title: string };

type SectionProps = { section: Section };
const renderSectionHeader = ({ section }: SectionProps) => (
  <SectionHeader title={section.title} />
);

const eventSections = (events: EventDays, locale: string): Section[] =>
  events.map(it => ({
    data: it,
    title: formatDate(it[0].fields.startTime[locale], "dddd D MMMM")
  }));

const eventIds = (events: EventDays): string[] =>
  events.map(day => day.map(e => e.sys.id)).reduce(concat, []);

class EventList extends Component<Props> {
  static defaultProps = {
    refreshing: false,
    onRefresh: undefined
  };

  shouldComponentUpdate(nextProps: Props) {
    const { locale, refreshing, savedEvents } = this.props;
    const {
      locale: nextLocale,
      refreshing: nextRefreshing,
      savedEvents: nextSavedEvents
    } = nextProps;

    const ids = eventIds(this.props.events);
    const nextIds = eventIds(nextProps.events);

    return (
      !equals(ids, nextIds) ||
      locale !== nextLocale ||
      refreshing !== nextRefreshing ||
      savedEvents !== nextSavedEvents
    );
  }

  render() {
    const {
      events,
      savedEvents,
      addSavedEvent,
      removeSavedEvent,
      locale,
      refreshing,
      onRefresh,
      onPress,
      getAssetUrl
    } = this.props;

    const isSavedEvent = id => savedEvents.has(id);

    return (
      <SectionList
        stickySectionHeadersEnabled
        sections={eventSections(events, locale)}
        renderSectionHeader={renderSectionHeader}
        renderSectionFooter={separator(styles.sectionFooter)}
        renderItem={renderItem({
          isSavedEvent,
          addSavedEvent,
          removeSavedEvent,
          locale,
          onPress,
          getAssetUrl
        })}
        keyExtractor={event => event.sys.id}
        contentContainerStyle={styles.container}
        ItemSeparatorComponent={separator(styles.itemSeparator)}
        SectionSeparatorComponent={separator(styles.sectionSeparator)}
        refreshing={refreshing}
        onRefresh={onRefresh}
      />
    );
  }
}

const styles = StyleSheet.create({
  itemSeparator: {
    height: 12
  },
  sectionSeparator: {
    height: 6
  },
  container: {
    paddingTop: 0,
    backgroundColor: bgColor
  },
  eventListItem: {
    borderRadius: 5,
    // The below properties are required for ioS shadow
    shadowColor: eventCardShadow,
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 1,
    shadowRadius: 3,
    // The below properties are required for android shadow
    borderWidth: 0,
    elevation: 3,
    backgroundColor: bgColor
  },
  sectionFooter: {
    height: 6
  }
});

export default EventList;
