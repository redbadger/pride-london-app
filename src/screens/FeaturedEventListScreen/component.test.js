// @flow
import React from "react";
import type { NavigationScreenProp } from "react-navigation";
import { shallow } from "enzyme";
import Component from "./component";
import EventList from "../../components/EventList";
import { EVENT_DETAILS } from "../../constants/routes";

const navigation: NavigationScreenProp<{
  params: { title: string }
}> = ({
  navigate: jest.fn(),
  state: {
    params: {
      title: "Events"
    }
  }
}: any);

describe("FeaturedEventListScreen Component", () => {
  it("renders correctly", () => {
    const output = shallow(
      <Component
        navigation={navigation}
        events={[]}
        getAssetUrl={() => ""}
        addSavedEvent={() => {}}
        removeSavedEvent={() => {}}
        savedEvents={new Set()}
      />
    );
    expect(output).toMatchSnapshot();
  });

  it("navigates to event details on event item press", () => {
    const output = shallow(
      <Component
        navigation={navigation}
        events={[]}
        getAssetUrl={() => ""}
        addSavedEvent={() => {}}
        removeSavedEvent={() => {}}
        savedEvents={new Set()}
      />
    );
    const onPress = output.find(EventList).prop("onPress");
    onPress("my-event");
    expect(navigation.navigate).toHaveBeenCalledWith(EVENT_DETAILS, {
      eventId: "my-event"
    });
  });
});

afterEach(() => {
  navigation.navigate.mockClear();
});
