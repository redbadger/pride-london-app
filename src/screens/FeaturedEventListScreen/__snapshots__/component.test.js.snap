// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedEventListScreen Component renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Header
    backgroundColor="#2d2f7f"
  >
    <ContentPadding
      padding={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <IconButton
        accessibilityLabel="Back"
        onPress={[Function]}
        source={1}
      />
      <Text
        color="blackColor"
        markdown={false}
        markdownStyle={Object {}}
        style={
          Object {
            "color": "#ffffff",
          }
        }
        type="h2"
      >
        Featured events
      </Text>
      <Component
        style={
          Object {
            "height": 48,
            "width": 48,
          }
        }
      />
    </ContentPadding>
  </Header>
  <EventList
    addSavedEvent={[Function]}
    events={Array []}
    getAssetUrl={[Function]}
    locale="en-GB"
    onPress={[Function]}
    refreshing={false}
    removeSavedEvent={[Function]}
    savedEvents={Set {}}
  />
</Component>
`;
